libraries/chain/transaction_context.cpp

```diff
@@ -113,13 +113,44 @@ namespace eosio { namespace chain {
          }
       }
 
+      // check for resource_payer extension
+      bool has_res_pyr = false;
+      resource_payer res_pyr;
+      if( trx.transaction_extensions.size() > 0 ) {
+          std::find_if( trx.transaction_extensions.begin(), trx.transaction_extensions.end(),
+                        [&](auto& elem) {
+                           bool ret = elem.first ==  transaction_extension_id::resource_payer_id;
+                           if (ret) {
+                              fc::raw::unpack(elem.second, res_pyr);
+                              has_res_pyr = true;
+                           }
+                           return ret;
+                        } );
+      }
+
```

Comment:
Victor added a method to get resource payer. Go ahead and use that when he commits it.

Reply:
Victor's resource_payer() method is getting only resource payer account name, where as I need the entire resource_payer struct. I'm adding an overloaded resource_payer() method, and moving it to transaction struct, so that both gettors are at one place for users to choose. This change is going in along with exceptions code in next PR.